File: content/atlas/source/includes/fts/field-types/boolean/create_index_basic.py
from pymongo import MongoClient
from pymongo.operations import SearchIndexModel
# Connect to your Atlas deployment
uri = "<connection-string>"
client = MongoClient(uri)
database = client["sample_analytics"]
collection = database["customers"]
# Create the MongoDB Search index definition for the geo field
search_index_model = SearchIndexModel(
    definition={
        "mappings": {
            "dynamic": False,
            "fields": {
                "active": {
                    "type": "boolean"
                }
            }
        },
    },
    name="default",
)
# Create the index
result = collection.create_search_index(model=search_index_model)
print(f"New index name: {result}")

File: content/atlas/source/includes/fts/field-types/boolean/create-index-basic.js
const { MongoClient } = require("mongodb");
// Replace the connection string with your MongoDB deployment's connection string
const uri =  "<connection-string>";
const client = new MongoClient(uri);
async function run() {
  try {
    const database = client.db("sample_analytics");
    const collection = database.collection("customers");
    // Create the MongoDB Search index definition for the boolean field
    const index = {
        name: "default",
        definition: {
            "mappings": {
                "dynamic": false,
                "fields": {
                    "active": {
                        "type": "boolean"
                    }
                }
            }
        }
    }
    // Create the index
    const result = await collection.createSearchIndex(index);
    console.log("New index name: " + result);
  } finally {
    await client.close();
  }
}
run().catch(console.dir);
